src/lt/plugins/auto_complete.cljs

```diff
@@ -148,20 +148,34 @@
 
 (declare hinter)
 
+(defn remove-long-completitions [hints]
+  (filter #(< (.-length (.-completion %)) (:hint-limit @hinter)) hints))
+
 (def hinter (-> (scmd/filter-list {:items (fn []
                                             (when-let [cur (pool/last-active)]
                                               (let [token (-> @hinter :starting-token :string)]
                                                 (distinct-completions
-                                                 (if token
-                                                   (remove #(= token (.-completion %))
-                                                           (object/raise-reduce cur :hints+ [] token))
-                                                   (object/raise-reduce cur :hints+ []))))))
+                                                 (remove-long-completitions
+                                                  (if token
+                                                    (remove #(= token (.-completion %))
+                                                            (object/raise-reduce cur :hints+ [] token))
+                                                    (object/raise-reduce cur :hints+ [])))
+                                                 ))))
                                    :key text|completion})
                 (object/add-tags [:hinter])))
 
 (defn on-line-change [line ch]
   (object/raise hinter :line-change line ch))
 
+(behavior ::set-hint-limit
+          :triggers #{:object.instant}
+          :type :user
+          :desc "Hinter: set maximum length of an autocomplete hint"
```

Comment:
This should probably be `:desc "Auto-complete: ...` – I'm not sure "hinter" would be widely understood.